.github/actions/label-closed-issues-with-date-v1/src/run.ts

```diff
@@ -0,0 +1,74 @@
+export default async function run(
+  core: typeof import('@actions/core'),
+  github: typeof import('@actions/github')
+): Promise<void> {
+  try {
+    // Get required inputs
+    const issueNumber = parseInt(core.getInput('issue-number', { required: true }))
+    const issueOrganization = core.getInput('issue-organization', { required: true })
+    const issueRepo = core.getInput('issue-repo', { required: true })
+    const token = core.getInput('token', { required: true })
+
+    core.info(`Checking issue ${issueNumber} in ${issueOrganization}/${issueRepo}`)
+
+    // Get the octokit instance
+    const octokit = github.getOctokit(token)
+
+    // Get issue details from REST API
+    const { data: issue } = await octokit.rest.issues.get({
+      owner: issueOrganization,
+      repo: issueRepo,
+      issue_number: issueNumber
+    })
+
+    core.info(`Issue state: ${issue.state}, closed_at: ${issue.closed_at}`)
+
+    // Check if issue is closed
+    if (issue.state === 'closed' && issue.closed_at) {
+      const closedDate = new Date(issue.closed_at)
+      const dateLabel = `Closed: ${closedDate.toISOString().split('T')[0]}` // Format as YYYY-MM-DD
+      
+      core.info(`Issue is closed. Adding date label: ${dateLabel}`)
+      
+      // Get current labels to find existing "Closed:" labels
+      const currentLabels = issue.labels || []
+      const closedLabelsToRemove = currentLabels
+        .filter((label) => typeof label === 'object' && label && 'name' in label && typeof label.name === 'string' && label.name.startsWith('Closed:'))
+        .map((label) => (label as { name: string }).name)
```

Comment:
This can be simplified: 

```suggestion
      const labelsToRemove: string[] = []
      for (const currentLabel of currentLabels) {
        const name =
          typeof currentLabel === 'string' ? currentLabel : currentLabel?.name

        if (name?.startsWith('Closed:')) {
          labelsToRemove.push(name)
        }
      }
```